[excel] (PivotTable) PivotLayout features (#2613)
OfficeDev/office-js-docs-pr: docs/excel/excel-add-ins-pivottables.md

![A PivotTable using the tabular layout.](../images/excel-pivots-tabular-layout.png)

### Other PivotLayout functions

> By default, PivotTables adjust row and column sizes as needed. This is done when the PivotTable is refreshed. `PivotLayout.autoFormat` specifies that behavior. Any row or column size changes made by your add-in persist when `autoFormat` is `false`. Additionally, the default settings of a PivotTable keep any custom formatting in the PivotTable (such as fills and font changes). Set `PivotLayout.preserveFormatting` to `false` to apply the default format when refreshed.

Reviewer: Out of curiosity, why didn't you include a link to [autoFormat](https://docs.microsoft.com/en-us/javascript/api/excel/excel.pivotlayout?view=excel-js-preview#autoFormat)?

Author: Mostly because the anchor links in the ref docs are going to change soon.

> A `PivotLayout`also controls header and total row settings, how empty data cells are displayed, and [alt text](https://support.microsoft.com/topic/add-alternative-text-to-a-shape-picture-chart-smartart-graphic-or-other-object-44989b2a-903c-4d9a-b742-6a75b451c669) options. The [PivotLayout](/javascript/api/excel/excel.pivotlayout) reference provides a complete list of these features.

Reviewer: Missing space 
```suggestion
A `PivotLayout` also controls header and total row settings, how empty data cells are displayed, and [alt text](https://support.microsoft.com/topic/add-alternative-text-to-a-shape-picture-chart-smartart-graphic-or-other-object-44989b2a-903c-4d9a-b742-6a75b451c669) options. The [PivotLayout](/javascript/api/excel/excel.pivotlayout) reference provides a complete list of these features.
```

> [!NOTE]
> Some of the PivotLayout functionality mentioned here is currently only available in public preview. [!INCLUDE [Information about using preview APIs](../includes/using-excel-preview-apis.md)]

> The following example makes empty data cells display the string "--", formats the body range to a consistent horizontal alignment, and ensures the formatting changes will stay after the PivotTable is refreshed.

Reviewer: 1. In a recent review for me, our editor changed all "example" references like this one (directly preceding a sample) to "code sample". The specificity of the phrasing helps with accessibility. 

2. Since it's referred to as a string, should "--" be wrapped in code ticks?

3. Consider switching simple perfect tense to present tense. 

```suggestion
The following code sample makes empty data cells display the string `"--"`, formats the body range to a consistent horizontal alignment, and ensures that the formatting changes remain even after the PivotTable is refreshed.
```

```js
Excel.run(function (context) {
    var pivotTable = context.workbook.pivotTables.getItem("Farm Sales");
    var pivotLayout = pivotTable.layout;

>     // This sets a default value for an empty cell in the PivotTable. This does not include cells left blank by the layout.

Reviewer: The comments in this code sample have some phrasing inconsistency. I recommend this strategy, but consistency in either direction would work as long as the grammar issues are resolved too. 
```suggestion
    // Set a default value for an empty cell in the PivotTable. This doesn't include cells left blank by the layout.
```

    pivotLayout.emptyCellText = "--";

    // Set the text alignment to match the rest of the PivotTable.
    pivotLayout.getDataBodyRange().format.horizontalAlignment = Excel.HorizontalAlignment.right;

>     // This turns ensures empty cells are filled with a default value.

Reviewer: ```suggestion
    // Ensure empty cells are filled with a default value.
```

    pivotLayout.fillEmptyCells = true;

>     // This ensure the PivotTable keeps any formatting that's been set after it is refreshed and recalculated.

Reviewer: I think "persist" will yield more clarity with localization. 
```suggestion
    // Ensure that the format settings persist, even after the PivotTable is refreshed and recalculated.
```

![A PivotTable using the tabular layout.](../images/excel-pivots-tabular-layout.png)

> #### LayoutType switch sample

Reviewer: ```suggestion
#### PivotLayout type switch code sample
```
